### ISSUE TITLE
Annotation to omit a table from migrations

### ISSUE THREAD

**Issue by rubensayshi** [facebook/ent#1079]
- [x] I have searched the [issues](https://github.com/facebook/ent/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

I was thinking that a simple extra `Annotation` property `IgnoreMigrations` would help anyone who for w/e reason wants to do some custom create statement for a table (or view).

## Motivation 🔦

We have a few views in our DB for which we'd like to use ent as well to query them, but currently ent will try to create them as tables during the migrate.

**Comment by a8m** [comment]
Hey @rubensayshi and thanks for proposing this option.

I was thinking about adding a Hook option to the migration, so users will have more control on this step. For example:

```go
err := client.Schema.Create(
    ctx,
    migrate.Hook(func(next schema.Creator) schema.Creator {
        return schema.CreateFunc(ctx context.Context, tables ...*schema.Table) error {
            // <filter tables here>
            return next.Create(ctx, tables...)
        })
    }),
)
```

Does it sound like something that can solve your issue?

**Comment by rubensayshi** [comment]
ye that sounds like a more flexible option indeed!

**Comment by a8m** [comment]
Closed with #1150 

Thanks @rubensayshi 😃
